sign/ed448/isogeny.go

```diff
@@ -0,0 +1,38 @@
+package ed448
+
+import fp "github.com/cloudflare/circl/math/fp448"
+
+// deg4isogeny is a 4-degree isogeny from Goldilocks curve to an isogenous
+// twisted Edwards curve with a=-1.
+// Goldilocks is x^2+y^2=1-39081x^2y^2
+// 4IsoCurve is -x^2+y^2=1-39082x^2y^2
+type deg4isogeny struct{}
+
+// Push sends a point in Goldilocks to a point in 4IsoCurve.
+func (m deg4isogeny) Push(p *pointR1) { m.calculate(p, false) }
+
+// Push sends a point in 4IsoCurve to a point in Goldilocks.
+func (m deg4isogeny) Pull(p *pointR1) { m.calculate(p, true) }
+
+func (m deg4isogeny) calculate(P *pointR1, aNegative bool) {
```

Comment:
I don't like this function. We should be able to use the type system to keep track of what curve we're on, and this breaks that idea